pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -51,10 +57,35 @@ func (t *ofTable) GetMissAction() MissActionType {
 	return t.missAction
 }
 
-func (t *ofTable) GetNext() TableIDType {
+func (t *ofTable) GetNext() uint8 {
 	return t.next
 }
 
+func (t *ofTable) SetNext(next uint8) {
+	t.next = next
+}
+
+func (t *ofTable) SetMissAction(action MissActionType) {
+	t.missAction = action
+}
+
+func (t *ofTable) SetBridge(bridge Bridge) {
+	br := bridge.(*OFBridge)
+	br.Lock()
+	defer br.Unlock()
+
+	br.tableCache[t.id] = t
+}
+
+func (t *ofTable) CreateOnBridge(next uint8, missAction MissActionType, bridge Bridge) {
+	t.next = next
+	t.missAction = missAction
+	br := bridge.(*OFBridge)
+	br.Lock()
+	defer br.Unlock()
+	br.tableCache[t.id] = t
```

Comment:
Why reverse the call? I feel the dependency will become unmanagable finally. ofTable is even operating bridge's lock. Can Bridge provide a method to register/create a table?
I also had the feeling when trying to update code in conjMatchFlowContext, conjMatchFlowContextChange, policyRuleConjunction, clause, client. Hard to understand their dependency map because there are many such reversed calls.

Reply:
This function tries to set the table's related properties in one function. I think you mean putting the following code block into a function in Bridge, and call that function in this one, right?
```
func (b *OFBridge) AddTable(t *ofTable) {
	b.Lock()
	defer br.Unlock()
	br.tableCache[t.id] = t
}
```